advanced/Scripts/query.sh

```diff
@@ -58,12 +61,131 @@ scanList(){
+    exit 1
+  else
+    # Check if the current database is more recent.
+    if [[ $(sqlite3 "${gravityOLDfile}" "SELECT value FROM info \
+              WHERE property IS 'updated';") \
+        > \
+          $(sqlite3 "${gravityDBfile}" "SELECT value FROM info \
+              WHERE property IS 'updated';") ]]; then
+      echo -e "  ${CROSS}  Old database ${gravityOLDfile} is more recent."
+      exit 1
+    fi
+    # Run the queries
+    printf %s "  ${INFO} Running queries, please be patient..."
+    # Find the status 1 queries first
+    mapfile -t domainarray < <( sqlite3 "${ftlDBfile}" << EOSQL
+      ATTACH '${gravityDBfile}' AS new;
+      ATTACH '${gravityOLDfile}' AS old;
+      SELECT DISTINCT queries.domain, '' || adlist.address || '' FROM queries
+      JOIN gravity ON new.gravity.domain = queries.domain
+      JOIN adlist ON adlist.id = new.gravity.adlist_id
+      WHERE queries.domain NOT IN (SELECT domain FROM old.gravity)
+      AND queries.domain IN (SELECT domain FROM new.gravity)
+      AND queries.status IN (1)
+      AND timestamp > (SELECT value FROM new.info WHERE property = 'updated');
+EOSQL
+      )
+    if [ ! "${#domainarray[@]}" -eq 0 ]; then
+      for (( j=0; j<"${#domainarray[@]}"; j++ )); do
+        IFS='|'
+        read -r domain urllist <<< "${domainarray[j]}"
+        printf '\r%s\n' "$(tput el)  ${INFO} Blocked: ${domain} (${COL_GREEN}${urllist}${COL_NC})"
+      done
+    fi
+    # Now find the CNAME matches (status 9)
+    mapfile -t cnamearray < <( sqlite3 "${ftlDBfile}" << EOSQL
+      ATTACH '${gravityDBfile}' AS new;
+      ATTACH '${gravityOLDfile}' AS old;
+      SELECT DISTINCT queries.domain, '' || adlist.address || '', queries.additional_info FROM queries
+      JOIN gravity ON new.gravity.domain = queries.additional_info
+      JOIN adlist ON adlist.id = new.gravity.adlist_id
+      WHERE queries.additional_info NOT IN (SELECT domain FROM old.gravity)
+      AND queries.additional_info IN (SELECT domain FROM new.gravity)
+      AND queries.status IN (9)
+      AND timestamp > (SELECT value FROM new.info WHERE property = 'updated');
+EOSQL
+      )
+    if [ ! "${#cnamearray[@]}" -eq 0 ]; then
+      for (( j=0; j<"${#cnamearray[@]}"; j++ )); do
+        IFS='|'
+        read -r domain urllist additional_info <<< "${cnamearray[j]}"
+        printf '\r%s\n' "$(tput el)  ${INFO} Blocked: ${domain}. (${COL_BLUE}CNAME${COL_NC})"
+        echo -e "        Domain: ${additional_info} (${COL_GREEN}${urllist}${COL_NC})"
+      done
+    fi
+    if [[ "${#domainarray[@]}" -eq 0 && "${#cnamearray[@]}" -eq 0 ]]; then
+      printf '\r%s\n' "$(tput el)  ${INFO} No new domains have been blocked yet."
+    fi
+  fi
+  exit 0
+}
+
 if [[ "${options}" == "-h" ]] || [[ "${options}" == "--help" ]]; then
     echo "Usage: pihole -q [option] <domain>
 Example: 'pihole -q -exact domain.com'
 Query the adlists for a specified domain
 
 Options:
+  -blocked            List blocked domains since last gravity run
```

Comment:
I wonder if the command should also have a trailing `new` as the other have. Just for consistency. This is open for debate, of course.
```suggestion
  -blockednew         List blocked domains since last gravity run
```